libusb.go

```diff
@@ -224,14 +224,27 @@ func (libusbImpl) setDebug(c *libusbContext, lvl int) {
 }
 
 func (libusbImpl) getDeviceDesc(d *libusbDevice) (*DeviceDesc, error) {
-	var desc C.struct_libusb_device_descriptor
+	var (
+		desc     C.struct_libusb_device_descriptor
+		pathData [8]uint8
+		path = []int{}
+		pathLen  int
+	)
 	if err := fromErrNo(C.libusb_get_device_descriptor((*C.libusb_device)(d), &desc)); err != nil {
 		return nil, err
 	}
+	if pathLen = int(C.libusb_get_port_numbers((*C.libusb_device)(d), (*C.uint8_t)(&pathData[0]), 8)); pathLen == 0{
+		pathLen = 1 //Root devices will have no path, increment pathLen to return port as 0
```

Comment:
the default value of `Port` will be 0, you don't need to set it explicitly. Setting `pathLen` to 1 here seems wrong, because the actual length of the path is 0. There is no guarantee that pathData[0] is 0, it may as well be complete garbage.

Reply:
That's fair - I hadn't considered that just because it returns correctly now that it's guaranteed to be zeroes.
Clarity > less code.